core/core-dotspacemacs.el

```diff
@@ -83,6 +83,10 @@ size to make separators look not too crappy.")
 By default the command key is `:' so ex-commands are executed like in Vim
 with `:' and Emacs commands are executed with `<leader> :'.")
 
+(defvar dotspacemacs-autosave-file-directly nil
+  "If non-nil, auto-save to the file directly. If nil, just save
+to auto-save file in `spacemacs-cache-directory'")
+
```

Comment:
Sorry for the delay to review this.

I did not understand the meaning of this variable, I only understood it by looking at where it is used in the code. We have te rename it.

`dotspacemacs-auto-save-new-buffer-to-file`